### PR TITLE
feat(stream): selective agg for stream

### PR DESCRIPTION
**Pull request by likg227**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

as title.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
related #3589

### CODE REVIEW

**Review by BowenXiao1999 — COMMENTED** [review]
Take a roughly look and seems good.

**Review by TennyZhuang — APPROVED** [review]
LGTM

File: src/common/src/array/bool_array.rs [hunk 1]
```diff
@@ -39,6 +40,10 @@ impl BoolArray {
         }
         builder.finish()
     }
+
+    pub fn to_bitmap(&self) -> Bitmap {
+        self.data.to_owned().bitand(self.null_bitmap())
```
Comment (xiangjinwu) [inline src/common/src/array/bool_array.rs:45]: nit: `&self.data & self.null_bitmap()`

File: src/stream/src/executor/aggregation/agg_state.rs [hunk 2]
```diff
@@ -29,6 +30,9 @@ pub struct AggState<S: StateStore> {
     /// Current managed states for all [`crate::executor::aggregation::AggCall`]s.
     pub managed_states: Vec<ManagedStateImpl<S>>,
 
+    /// Filters of `agg_call`s
+    pub filters: Vec<Option<ExpressionRef>>,
+
```
Comment (xiangjinwu) [inline src/stream/src/executor/aggregation/agg_state.rs:35]: These should not be part of the state.

**Review by StrikeW — APPROVED** [review]
LGTM

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**